java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureConsumerRebalancerListener.java

```diff
@@ -40,7 +49,13 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
       Optional<AzureOffsetMarker> partitionOffset = topicPartitionOffsetProvider.getOffset(partitionKey);
       partitionOffset.ifPresentOrElse(
           offset -> kafkaConsumer.seek(partition, offset.getOffsetValue()),
-          () -> kafkaConsumer.seekToBeginning(Collections.singletonList(partition)));
+          () -> {
+            if (shouldSeekToLatest) {
```

Comment:
small thing, an optimisation to collect all the topic-partitions to seek at the beginning or end and have one call to kafka.

Reply:
Good one, I implemented it!